src/nvm_vblk.c

```diff
@@ -576,23 +576,14 @@ static inline ssize_t vblk_sync_pread_s20(struct nvm_vblk *vblk, void *buf,
 
 	const int NTHREADS = NVM_MIN(nchunks, nsectr / WS_OPT);
 
-	if (nsectr % WS_OPT) {
```

Comment:
True, alignment to WS_OPT / the stripe_unit should not be required for reads.
However, alignment to sector-size is probably a good idea, I will have a closer look, but it should probably be changed, instead of removed, to something like:

```
	if (offset % sectr_nbytes) {
		NVM_DEBUG("FAILED: unaligned offset: %zu", offset);
		errno = EINVAL;
		return -1;
	}
	if (count % sectr_nbytes) {
		NVM_DEBUG("FAILED: unaligned count: %zu", count);
		errno = EINVAL;
		return -1;
	}
```

Reply:
Ah, yes that is right. I added your suggestion.